app/assets/javascripts/screen/screenController.js

```diff
@@ -28,4 +28,32 @@ angular.module('ocWebGui.screen', ['ui.router', 'ngResource', 'ocWebGui.shared.t
         });
 
         fetchData();
+        
+        // Agent status and colors classifications        
+        $scope.agents = $resource('agents.json').query(function() {
```

Comment:
Data fetching should be done in `fetchData` function. Also assignment to `$scope.agents` should happen only inside the query callback.